tests/base/boring_model.py

```diff
@@ -53,9 +56,24 @@ def __getitem__(self, index):
     def __len__(self):
         return self.len
 
+def count_calls(fn: Callable) -> Callable:
```

Comment:
why are you mimicking unittest mock, we already use it elsewhere...

Reply:
Not really. Mocking will remove the function I need. Here I am just counting how many time it got called.